src/seq.rs

```diff
@@ -13,33 +13,33 @@
 use super::Rng;
 use std::collections::hash_map::HashMap;
 
-/// Randomly sample *up to* `amount` elements from a finite iterator.
+/// Randomly sample `amount` elements from a finite iterator.
 ///
-/// The values are non-repeating but the order of elements returned is *not* random.
+/// The following can be returned:
+/// - `Ok`: `Vec` of `amount` non-repeating randomly sampled elements. The order is not random.
+/// - `Err`: `Vec` of *less than* `amount` elements in sequential order. This is considered an
+///   error since exactly `amount` elements is typically expected.
```

Comment:
For the `Err` case I'd prefer more precise documentation: in case less than `amount` elements are available, a `Vec` of exactly the elements in `iterable` is returned. The order is not random.

Reply:
the order is actually guaranteed to be **sequential**. I've updated the doc, let me know if that is better.